### DISCUSSION TITLE
Use Turbo engine to speed up the Rust compiler

### DISCUSSION THREAD

**Discussion by Boscop** [vercel/turborepo#2841] [Ideas]
I wish the Turbo engine could be used to speed up the Rust compiler, so that Rustaceans can also benefit from this approach :)
It would be very useful for shortening the edit/compile/run cycle of Rust executables and Rust wasm frontends!

The way it would work is: When changing the code of a symbol's definition in a large Rust project, only that symbol (or the containing file) and other symbol definitions that depend on it have to be recompiled.
Also the linking could work in a similar lazy/FRP/caching way.
It would be used instead of `cargo watch -x run` and the compiler would keep the symbol graph cache in memory and persist it to disk for reuse in future sessions.
(And later, IDEs could send queries about symbols to the running compiler to get all kinds of semantic info.)

What would need to be done to make this possible, in regards to the code architecture of the Turbo engine and the Rust compiler?

**Comment by anthonyshew** [comment]
Turbopack and the Turbo Engine is currently scoped to web technologies. The Rust compiler is highly optimized with many optimizations that are reminiscent of what you're describing out of the box!
